routes/routes.js

```diff
@@ -0,0 +1,39 @@
+const express = require('express');
+const router = express.Router();
+
+const passport = require('../helpers/passportInfo');
+const loginBlock = require('../helpers/loginBlock');
+
+// controllers file
+const homeController = require('../controllers/homeController.js');
+const authController = require('../controllers/authController.js');
+const userController = require('../controllers/userController.js');
+
+// routes available without login
+router.get('/', homeController.home);
+router.post('/search', homeController.search);
+
+// user authentication
+router.get('/login', authController.login);
+router.post('/login',
+  passport.authenticate('local', {
+    successRedirect: '/',
+    failureRedirect: '/login',
+    successFlash: 'Welcome!',
+    failureFlash: true
+  }));
+router.get('/register', authController.register);
+router.post('/register', authController.signup);
+router.get('/logout', authController.logout);
+
+// routes available with login
+router.get('/profile/:id', loginBlock.isLoggedIn, userController.profile);
+router.post('/profile/:id', loginBlock.isLoggedIn, userController.change);
+router.get('/result/:id', loginBlock.isLoggedIn, userController.result);
+router.delete('/delete/search/:id', userController.deleteSearch);
```

Comment:
you should check `DELETE` and `PUT` request too whether the user has been authenticated or not, not only the `GET` and `POST` request